src/Database/Adapter/PdoAdapter.php

```diff
@@ -403,6 +409,50 @@ private function isMulti(array $data): bool
         return true;
     }
 
+    private function splitColumnNameAndOperator($key, $value, &$columnName, &$operator): void
+    {
+        // initialize both column name and operator
+        $columnName = $key;
+        $operator = '=';
+
+        // check presence of operator in $key
+        if (preg_match('/^(.*)\s(=|!=|<>|<|<=|>|>=)$/', $key, $matches) === 1) {
+            $columnName = $matches[1];
+            $operator = $matches[2];
+        }
+
+        $columnName = trim($columnName);
+
+        if (is_null($value)) {
+            if ($operator === '=') {
+                $operator = 'IS';
+                return;
+            }
+            if (in_array($operator, [ '!=', '<>' ])) {
+                $operator = 'IS NOT';
+                return;
+            }
+            throw new UnexpectedValueException('Cannot accept "' . $operator . '" operator for NULL value');
+        }
+
+        if (is_array($value)) {
+            if ($operator === '=') {
+                $operator = 'IN';
+                return;
+            }
+            if (in_array($operator, [ '!=', '<>' ])) {
+                $operator = 'NOT IN';
+                return;
+            }
+            throw new UnexpectedValueException('Cannot accept "' . $operator . '" operator for list value');
+        }
+        
+        // always prefer '<>' to '!='
+        if ($operator === '!=') {
```

Comment:
Is this any sql recommendation?

Reply:
[https://stackoverflow.com/a/13740785/9228492](https://stackoverflow.com/a/13740785/9228492) actually can't find any official documentation

Reply:
ok thanks :)